src/ripple/beast/core/CurrentThreadName.cpp

```diff
@@ -0,0 +1,149 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2013, Vinnie Falco <vinnie.falco@gmail.com>
+
+    Portions of this file are from JUCE.
+    Copyright (c) 2013 - Raw Material Software Ltd.
+    Please visit http://www.juce.com
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/beast/core/CurrentThreadName.h>
+#include <boost/thread/tss.hpp>
+
+namespace beast {
+namespace detail {
+
+static boost::thread_specific_ptr<std::string> threadName;
+
+void saveThreadName (std::string name)
+{
+    try
+    {
+        threadName.reset (new std::string {std::move(name)});
+    }
+    catch(...)
+    {
+    }
+}
+
+} // detail
+
+std::string getCurrentThreadName ()
+{
+    return detail::threadName.get() ? *detail::threadName.get() : "";
+}
+
+} // beast
+
+//------------------------------------------------------------------------------
+
+#if BEAST_WINDOWS
+
+#include <windows.h>
+#include <process.h>
+#include <tchar.h>
+
+namespace beast {
+namespace detail {
+
+void setCurrentThreadNameImpl (std::string const& name)
+{
+   #if BEAST_DEBUG && BEAST_MSVC
+    struct
+    {
+        DWORD dwType;
+        LPCSTR szName;
+        DWORD dwThreadID;
+        DWORD dwFlags;
+    } info;
+
+    info.dwType = 0x1000;
+    info.szName = name.c_str ();
+    info.dwThreadID = GetCurrentThreadId();
+    info.dwFlags = 0;
+
+    __try
+    {
+        RaiseException (0x406d1388 /*MS_VC_EXCEPTION*/, 0, sizeof (info) / sizeof (ULONG_PTR), (ULONG_PTR*) &info);
```

Comment:
I do not understand the purpose of this code. It looks like we raise a structured exception and then catch it and continue (right?). I'm clearly missing something.